java/src/jmri/jmrit/symbolicprog/VariableValue.java

```diff
@@ -512,17 +668,19 @@ protected int setValueInCV(int oldCv, int newVal, String maskString, int maxVal)
         } else {
             int radix = Integer.parseInt(maskString);
             int lowPart = oldCv % radix;
-            int newPart = newVal%(maxVal+1)*radix;
-            int highPart = (oldCv / (radix*(maxVal+1)) ) * (radix * (maxVal+1));
-            log.trace("Set sees oldCv {} radix {}, lowPart {}, newVal {}, highPart {}, does {}", oldCv, radix, lowPart, newVal, highPart, highPart+newVal*radix+lowPart);
-            return highPart+newPart+lowPart;
+            int newPart = newVal % (maxVal + 1) * radix;
+            int highPart = (oldCv / (radix * (maxVal + 1))) * (radix * (maxVal + 1));
```

Comment:
Is the value calculated in this trace correct, or should it match what is returned by this method?

Reply:
Thanks.

Those particular changes were simply the result of running the NetBeans->Format command. Has it made a code change (it doesn't seem so to me) or was the problem you have pointed out pre-existing. I think the latter.

It's 3:05 am here and I need to go back to bed and get some more sleep - my mind doesn't want to wrap around this at the moment. Would you like to suggest what line 681 should read?

Until the morning...

Reply:
Git Blame says @bobjacobsen changed that trace 10 months ago https://github.com/JMRI/JMRI/commit/8050c73e99e2849a5ab4cc222ba8da108c4ade90 so I'll pass to him for comment.

Reply:
Here's the original trace line:
```
% git diff 7bc80e0..8050c73e99e284 java/src/jmri/jmrit/symbolicprog/VariableValue.java
...
@@ -466,7 +467,8 @@ public abstract class VariableValue extends AbstractValue implements java.beans.
         } else {
             int radix = Integer.parseInt(maskString);
             int lowPart = oldCv % radix;
-            int highPart = (oldCv / (radix * maxVal))*(radix * maxVal);
+            int highPart = (oldCv / (radix*(maxVal+1)) ) * (radix * (maxVal+1));
+            log.trace("Set sees oldCv {} radix {}, lowPart {}, newVal {}, highPart {}, does {}", oldCv, radix, lowPart, newVal, highPart, highPart+newVal*radix+lowPart);
             return highPart+newVal*radix+lowPart;
         }
```

The trace line is showing the same calculation as what's about to be returned. That different here.  Looks like a cut&paste error somewhere after.

Reply:
Looks like this one did it: https://github.com/JMRI/JMRI/commit/ca35f4ab3b2198bcc03b03f33d1065f8bc71cbc0

Reply:
@bobjacobsen I'd like to align the trace and return values before merging this, but have no idea which to change...